propertyestimator/properties/enthalpy.py

```diff
@@ -2,154 +2,290 @@
+        pass
+
+    def __init__(self, protocol_id):
+        super().__init__(protocol_id)
+
+        self._value = None
+        self._component = None
+        self._full_substance = None
+
+        self._weighted_value = None
+
+    def execute(self, directory, available_resources):
+
+        assert len(self._component.components) == 1
+
+        main_component = self._component.components[0]
+        mole_fraction = 0.0
+
+        for component in self._full_substance.components:
+
+            if component.smiles != main_component.smiles:
+                continue
+
+            mole_fraction = component.mole_fraction
+
+        self._weighted_value = mole_fraction * self.value
+
+        if not isinstance(self._weighted_value, unit.Quantity):
+            self._weighted_value = unit.Quantity(self._weighted_value, None)
+
+        return self._get_output_dictionary()
 
 
 @register_estimable_property()
-class Enthalpy(PhysicalProperty):
-    """A class representation of a enthalpy property"""
+@register_thermoml_property(thermoml_string='Excess molar enthalpy (molar enthalpy of mixing), kJ/mol')
+class EnthalpyOfMixing(PhysicalProperty):
+    """A class representation of an enthalpy of mixing property"""
+
+    EnthalpyWorkflow = namedtuple('EnthalpySchema', 'build_coordinates '
+                                                    'assign_topology '
+                                                    'energy_minimisation '
+                                                    'npt_equilibration '
+                                                    'npt_production '
+                                                    'extract_enthalpy')
 
     @staticmethod
-    def get_default_calculation_schema():
+    def get_enthalpy_workflow(id_prefix=''):
+        """Returns the set of protocols which when combined in a workflow
+        will yield the enthalpy of a substance.
 
-        schema = WorkflowSchema(property_type=Enthalpy.__name__)
-        schema.id = '{}{}'.format(Enthalpy.__name__, 'Schema')
+        Parameters
+        ----------
+        id_prefix: str
+            A prefix to append to the id of each of the returned protocols.
 
-        # Initial coordinate and topology setup.
-        build_coordinates = protocols.BuildCoordinatesPackmol('build_coordinates')
+        Returns
+        -------
+        EnthalpyOfMixing.EnthalpyWorkflow
+            The protocols used to estimate the enthalpy of a substance.
+        """
 
-        build_coordinates.substance = ProtocolPath('substance', 'global')
+        build_coordinates = protocols.BuildCoordinatesPackmol(id_prefix + 'build_coordinates')
 
-        schema.protocols[build_coordinates.id] = build_coordinates.schema
+        build_coordinates.substance = ProtocolPath('substance', 'global')
 
-        assign_topology = protocols.BuildSmirnoffTopology('build_topology')
+        assign_topology = protocols.BuildSmirnoffTopology(id_prefix + 'build_topology')
```

Comment:
What do you think about renaming this `BuildSmirnoffSystem`? The `Topology` object is a specific object in the basic `openforcefield` toolkit, and I don't think it has parameters so I'm worried this name could generate confusion.

Reply:
I agree!